Islandora/islandora
# Media multifile

# What does this Pull Request do?

Adds the ability to create derivative files to attach to existing media.

# What's new?
 A MediaSourceHasMimetype condition to allow media-level behaviours based on a Media's sourcefile's mimetype.
 A new Reaction to generate files with microservices, and have them attach to existing media.
 A new Action to define which queue the file is sent to, and where the returned derivative is to be attached.
 A new action to both attach an extracted text file to an existing media, and populate an editable text field with the contents of that file.
 A modification to islandora_entity_view_mode_alter to allow media views to be altered as well as node views.

# How should this be tested?
 With this patch in place all current Islandora Actions, Reactions and Contexts should continue to function as they have in the past.
 A user should be able to add a file field to an existing media type, create an Action to populate that field, and create a context to cause that Action to fire.
[ Demo here]( https://youtu.be/3U6tBvD8oJY).  (Rough cut full of typos and stumbling, but you'll get the idea)


# Additional Notes:
I'll put a small features module together over the course of the next few days to make a quick evaluation a little easier

Example:
* Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation? 
* Does this change add any new dependencies? 
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
* Could this change impact execution of existing code?

# Interested parties
@Islandora-Devops/committers
